integrations/kafka_connector/src/main/java/software/amazon/timestream/utility/TimestreamWriter.java

```diff
@@ -0,0 +1,313 @@
+package software.amazon.timestream.utility;
+
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.core.exception.SdkException;
+import software.amazon.awssdk.services.timestreamwrite.model.Record;
+import software.amazon.awssdk.services.timestreamwrite.model.*;
+import software.amazon.timestream.table.schema.DimensionMapping;
+import software.amazon.timestream.table.schema.MultiMeasureAttributeMapping;
+import software.amazon.timestream.table.schema.SchemaDefinition;
+import software.amazon.timestream.TimestreamSinkConnectorConfig;
+import software.amazon.timestream.TimestreamSinkConstants;
+
+import java.time.Instant;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Class that receives the non-empty Kafka messages as {@link SinkRecord}
+ * objects and writes to Timestream table as records
+ */
+public class TimestreamWriter {
+
+    /**
+     * Logger object
+     */
+    private static final Logger LOGGER = LoggerFactory.getLogger(TimestreamWriter.class);
+    /**
+     * Timestream database name
+     */
+    private final String databaseName;
+    /**
+     * Timestream table name
+     */
+    private final String tableName;
+    /**
+     * Version will be added to record if enabled
+     */
+    private final boolean enableVersioning;
+    /**
+     * Empty dimension value is skipped if enabled
+     */
+    private final boolean skipDimension;
+    /**
+     * Empty measure value is skipped if enabled
+     */
+    private final boolean skipMeasure;
+    /**
+     * Offset tracker per partition
+     */
+    private final Map<String, Set<Long>> recordOffsets = new ConcurrentHashMap<>();
+    /**
+     * Offset tracker set
+     */
+    private final Set<Long> recordOffsetsList = new HashSet<>();
+    /**
+     * Timestream table schema definition
+     */
+    private final SchemaDefinition schemaDefinition;
+
+    /**
+     *
+     * @param schemaDefinition table schema
+     * @param config connector config
+     */
+    public TimestreamWriter(final SchemaDefinition schemaDefinition, final TimestreamSinkConnectorConfig config) {
+
+        this.databaseName = config.getDatabaseName();
```

Comment:
Can you also capture the partition key on Timestram table? https://docs.aws.amazon.com/timestream/latest/developerguide/customer-defined-partition-keys.html and when you create the table ( may be during initialization ), you  can create with that specific partition key

Reply:
Sure, created the Timestream table with partition key (as part of the CFT) and added a statement here in the log that mentions the partition key